hooks/terraform_docs.sh

```diff
@@ -218,8 +251,10 @@ function terraform_docs {
       rm -f "$tmp_file_docs_tf"
     fi
 
+    # Use of insertion markers to insert the terraform-docs output between the markers
     # Replace content between markers with the placeholder - https://stackoverflow.com/questions/1212799/how-do-i-extract-lines-between-two-line-delimiters-in-perl#1212834
-    perl -i -ne 'if (/BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK/../END OF PRE-COMMIT-TERRAFORM DOCS HOOK/) { print $_ if /BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK/; print "I_WANT_TO_BE_REPLACED\n$_" if /END OF PRE-COMMIT-TERRAFORM DOCS HOOK/;} else { print $_ }' "$text_file"
+    perl_expression="if (/$insertion_marker_begin/../$insertion_marker_end/) { print \$_ if /$insertion_marker_begin/; print \"I_WANT_TO_BE_REPLACED\\n\$_\" if /$insertion_marker_end/;} else { print \$_ }"
```

Comment:
Given this improvement, I'd also suggest to format it to make it readable. Something like below (needs testing as I'm writing from the top of my head):
```suggestion
    read -d '' perl_expression <<-EOD
        if (/$insertion_marker_begin/../$insertion_marker_end/) {
            print \$_ if /$insertion_marker_begin/
            print \"I_WANT_TO_BE_REPLACED\\\n\$_\" if /$insertion_marker_end/
        } else {
            print \$_
        }
EOD
```

Caveats:
- `\\n` becomes `\\\n`
- closing `EOD` has to have no indentation

Unrelated: this might be an overengineering, though I'd enclose markers search patterns into caret (`^`) and dollar (`$`) signs, like below:
```shell
    read -d '' perl_expression <<-EOD
        if (/^$insertion_marker_begin\$/../^$insertion_marker_end\$/) {
            print \$_ if /^$insertion_marker_begin\$/
            print \"I_WANT_TO_BE_REPLACED\\\n\$_\" if /^$insertion_marker_end\$/
        } else {
            print \$_
        }
EOD
```

Reply:
I think I'll leave this to someone who knows perl 🤣

Reply:
Up to you. Though formatting suggestion above (single to multi line) doesn't make any changes in Perl's expression.